tracing/src/main/java/com/palantir/tracing/Traces.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright 2016 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.tracing;
+
+import com.google.common.base.Optional;
+import com.google.common.collect.Sets;
+import java.util.ArrayDeque;
+import java.util.Deque;
+import java.util.Set;
+
+public final class Traces {
+
+    public interface Headers {
+        String TRACE_ID = "X-B3-TraceId";
+        String PARENT_SPAN_ID = "X-B3-ParentSpanId";
+        String SPAN_ID = "X-B3-SpanId";
+        String SPAN_NAME = "X-Span-Name";
+    }
+
+    // stack of trace states
+    private static final ThreadLocal<Deque<TraceState>> STATE = new ThreadLocal<Deque<TraceState>>() {
```

Comment:
Yeah, you can add spans within an application using the static derive/complete stuff, which would get you more fine-grained detail of what's happening if you emit stuff to, say, Zipkin.